core/src/avm2/globals/XML.as

```diff
@@ -209,6 +215,10 @@ package {
             return "XML";
         };
 
+        prototype.hasOwnProperty = function(prop:String):Boolean {
+            return this.hasOwnProperty(prop);
```

Comment:
Just checking- this doesn't need the `self.AS3::` logic, right? Could you add a comment noting that?

Reply:
If I were to guess it probably does not need it as the `hasOwnProperty`method is not in AS3 namespace.